### PR TITLE
Add SysName to Oxidized view

### PR DESCRIPTION
**Pull request by smiles1969**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 10012`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Hi @smiles1969 
Thanx for your PR. This patch can make sense in some situation, but not really if you have "force_ip_to_sysname" enabled in the LibreNMS instance. In that case you end up with 2times the sysname displayed, which is a little bit annoying : 

Capture d’écran 2019-03-23 à 00 55 42

May be you could replace this column by the IP address in case the "force_ip_to_sysname" config option is set. Or make the column hidden by default, so it is only enabled when needed.

**Comment by smiles1969** [comment]
I have added the config checks as suggested

File: html/includes/functions.inc.php [hunk 1]
```diff
@@ -1349,6 +1349,15 @@ function get_oxidized_nodes_list()
         }
         echo "
         </td>
+        ";
+        if ($config['force_ip_to_sysname'] === false) {
```
Comment (murrant) [inline html/includes/functions.inc.php]: Please use `Config::get('force_ip_to_sysname')` instead of accessing the global variable.

Comment (smiles1969) [reply html/includes/functions.inc.php]: Done

**Comment by murrant** [comment]
@smiles1969 sorry, I missed this the first time but I just noticed you used strict comparison.  A loose comparison would be better as the value may be null.  Or simply just `!Config::get('force_ip_to_sysname')`

**Comment by smiles1969** [comment]
I've changed to !Config::get instead of == loose comparison and tested on my local repository.

**Comment by PipoCanaja** [comment]
@smiles1969 Could you have a look at Travis tests ?

**Comment by PipoCanaja** [comment]
Looks good. @librenms/reviewers  if you have a test setup with oxidized and not force_ip_to_sysname, do not hesitate to give feedback.

**Comment by murrant** [comment]
@smiles1969 thanks for all the changes, but I actually realized something when looking at this today.
We should always send the column data and have it exist, just hide it when appropriate

```
echo '<th data-column-id="sysname" data-visible="' . (!Config::get('force_ip_to_sysname')  ? 'true' : 'false') . '">SysName</th>';
```

File: html/includes/functions.inc.php [hunk 2]
```diff
@@ -1349,6 +1349,15 @@ function get_oxidized_nodes_list()
         }
         echo "
         </td>
+        ";
+        if (!Config::get('force_ip_to_sysname')) {
```
Comment (murrant) [inline html/includes/functions.inc.php]: There should be no if statement here, just output the column

File: html/pages/oxidized.inc.php [hunk 3]
```diff
@@ -30,6 +30,8 @@
                             <thead>
                             <tr>
                                 <th data-column-id="hostname" data-order="desc">Hostname</th>
+                                <?php echo '<th data-column-id="sysname" data-visible="' . (!Config::get('force_ip_to_sysname')  ? 'true' : 'false') . '">SysName</th>';
```
Comment (murrant) [inline html/pages/oxidized.inc.php]: you could move most of the html outside the php tags, just the true/false ternary needs to be inside <?php

**Comment by murrant** [comment]
Looks good :)

**Pull request merged**